### PR TITLE
Tiny JS World

### PR DESCRIPTION
**Pull request by zophrox**
### Building a Tiny JS World

[Demo](https://zophrox.github.io/zophrox.a-tiny-JS-world/) |
[Code base](https://github.com/zophrox/zophrox.a-tiny-JS-world/blob/gh-pages/index.js)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/zophrox/tiny/index.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+var dog = {kind:"animal", who:"dog", legs:"4", hands:"0", name:"Rex" ,gender:"male" , say:"woof-woof!", friend:"Mary, Jon"};
+var cat = {kind:"animal", who:"cat", legs:"4", hands:"0", name:"Lisa" ,gender:"female" , say:"meow!"};
```
Comment (OleksiyRudenko) [inline submissions/zophrox/tiny/index.js]: Please, format object literals so they read better, i.e. each property has its own line.
This is important as we write code for other people, firsthand.

File: submissions/zophrox/tiny/index.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+var woman = {kind:"human", who:"woman",legs:"2", hands:"2", name:"Mary" ,gender:"female" , say:"Hello Jon!", friend:"Mary, Rex"};
+var man = {kind:"human", who:"man",legs:"2", hands:"2", name:"Jon" ,gender:"male" , say:"Hello Mary!", friend:"Jon, Rex"};
+var  inhabitantes = [dog, cat, woman, man];
+inhabitantes.forEach(function(element){
+    var person = "";
+    person += element.kind + ";" + element.who + ";" + element.legs + ";"
+    + element.hands + ";" + element.name + ";" + element.gender + ";" + element.say;
```
Comment (OleksiyRudenko) [inline submissions/zophrox/tiny/index.js]: Please, find an appropriate method among [`Array` methods](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array) to join values.

File: submissions/zophrox/tiny/index.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    if(element.friend !== undefined){
+        person += "; " + element.friend;
+    }
+    console.log(person);
```
Comment (OleksiyRudenko) [inline submissions/zophrox/tiny/index.js]: Please, use `print` for output as per instructions.

Comment (OleksiyRudenko) [reply submissions/zophrox/tiny/index.js]: Please, use [ternary operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator) to select what to include into the string based on `.friend` value.

Comment (zophrox) [reply submissions/zophrox/tiny/index.js]: can i use a commits? Should there be only one commit?

Comment (OleksiyRudenko) [reply submissions/zophrox/tiny/index.js]: As many as you need.

**Comment by zophrox** [comment]
Everything is fixed

File: submissions/zophrox/tiny/index.js (hunk 3 above)
Comment (zophrox) [reply submissions/zophrox/tiny/index.js]: Everything is fixed

**Comment by OleksiyRudenko** [comment]
@zophrox Please, resolve conversations where you fixed things.

If any conversation contains a question, please, respond in its thread.

If you decided not to apply recommendations, please, give a reason in relevant thread.

The above builds up communication and gives idea on what you have worked on and what is yet to work with, without looking into the code and comparing versions.

**Comment by zophrox** [comment]
> @zophrox Please, resolve conversations where you fixed things.
> 
> If any conversation contains a question, please, respond in its thread.
> 
> If you decided not to apply recommendations, please, give a reason in relevant thread.
> 
> The above builds up communication and gives idea on what you have worked on and what is yet to work with, without looking into the code and comparing versions.
Ok, sorry

File: submissions/zophrox/tiny/index.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            say:"Hello Mary!",
+            friend:"Jon, Rex"};
+
+var  inhabitantes = [dog, cat, woman, man];
+
+inhabitantes.forEach(function(element){
+    var personArr = [];
+    var personStr = "";
+    personArr.push(element.kind, element.who, element.legs, element.hands, element.name, element.gender, element.say);
+    personStr = personArr.join(";");
```
Comment (OleksiyRudenko) [inline submissions/zophrox/tiny/index.js]: You can use array literal with variables inside. E.g. `var persons = [element.kind, element.who, etc...]`
And also chain array methods immediately. E.g.
```
var someString = ['abc', someVar, anotherVar]
  .join('; ') 
  + anythingElse;
```

Comment (zophrox) [reply submissions/zophrox/tiny/index.js]: Maybe I'm wrong. But in my opinion, my decision is more clear, at least for me. I do not mind to redo, if necessary

Comment (OleksiyRudenko) [reply submissions/zophrox/tiny/index.js]: Compare
```
const someArr;
someArr.push('a', 'b', 'c', 'd');
```
vs
```
const someArr = ['a' , 'b', 'c', 'd'];
```

The former is too verbose.
The latter is not my personal preference but rather a convention.
We write code for other people, firsthand. Studying is a good time to adopt conventions.

A much similar story like with objects specifications. You may feel comfortable with a one-liner. However, in commercial development you will have to follow conventions that are result of 1000's of professional discussions.

Chaining was also invented as a response to demand for better readable and shorter code.

Perfect function doesn't need any variables.

Please, fix to be able to follow the conventions if you are required to.

**Comment by zophrox** [comment]
ok, thanks


<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Вірусів
немає. www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

пт, 28 груд. 2018 о 22:49 Oleksiy Rudenko <notifications@github.com> пише:

**Comment by zophrox** [comment]
sorry, I accidentally clicked "resolve conversation"


<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Вірусів
немає. www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

пт, 28 груд. 2018 о 22:52 Oleg Kuntyy <zophrox@gmail.com> пише:

**Pull request merged**